BarrysPeanutsTemporalJava/src/main/java/barryspeanuts/ShoppingCartWorkflowImpl.java

```diff
@@ -0,0 +1,76 @@
+package barryspeanuts;
+
+import barryspeanuts.model.CreditCard;
+import barryspeanuts.model.Purchase;
+import barryspeanuts.model.PurchaseItem;
+import io.temporal.workflow.Workflow;
+import io.temporal.workflow.WorkflowQueue;
+import java.util.ArrayList;
+import java.util.List;
+import org.slf4j.Logger;
+
+public class ShoppingCartWorkflowImpl implements ShoppingCartWorkflow {
+  private static final Logger logger = Workflow.getLogger(ShoppingCartWorkflowImpl.class);
+
+  boolean exit = false;
+  List<PurchaseItem> purchaseItems = new ArrayList<>();
+  private final WorkflowQueue<Runnable> queue = Workflow.newWorkflowQueue(1024);
+
+  @Override
+  public void startWorkflow() {
+
+    logger.info("Starting Workflow for Barry's Peanuts");
+    while (true) {
+      Workflow.await(() -> exit);
+    }
+  }
+
+  @Override
+  public List<PurchaseItem> queryPurchaseItems() {
+    return this.purchaseItems;
+  }
+
+  @Override
+  public void addItem(PurchaseItem purchaseItem) {
+    this.purchaseItems.add(purchaseItem);
+  }
+
+  @Override
+  public void removeItem(PurchaseItem purchaseItem) {
+    this.purchaseItems.remove(purchaseItem);
+  }
+
+  @Override
+  public void checkOut(Purchase purchase) {
+
+    logger.info("Checking out purchase id {}", purchase.getId());
+  }
+
+  @Override
+  public void pay(Purchase purchase, CreditCard creditCard) {
```

Comment:
Why does checkout and pay need the purchase items in the purchase? it already has the items on the class.

Reply:
I did this to make it so the Temporal Server passes state information to the method. Maybe I don't have a correct understanding, but my thinkings is that if the machine running the Worker/Worflow goes down the state purchase items is lost. My intent was to delegate state persistence to the Temporal Server @cretz . But, then again, maybe I am not understanding things correctly.

Reply:
> Maybe I don't have a correct understanding, but my thinkings is that if the machine running the Worker/Workflow goes down the state of the purchase items is lost.

This understanding is incorrect. The entire magic of Temporal is we replay to build that state back up. This is one of the main things to show as different from Akka. Workflows can be stateful and that state is durable (another worker resuming the workflow will have that state rebuilt properly).